pkg/applicationmanager/controllers/applicationrestore.go

```diff
@@ -659,11 +659,26 @@ func (a *ApplicationRestoreController) restoreVolumes(restore *storkapi.Applicat
 		logrus.Errorf("error in checking backuplocation type")
 		return err
 	}
+
+	// Flag onlyPXDdriver to avoid creating a nfs-restore-pvc job for PXD driver(only) exclusively.
+	// This code checks if there's precisely one entry in the backupVolumeInfoMappings map.
+	// If such an entry exists, it sets onlyPXDdriver to true if "pxd" is the only driver in the map.
+	// If there are multiple drivers with PXD, we can still create a resourceExport CR as the count of restored volumes doesn't match the total count of PVCs to be restored.
+	// If there's only PXD with NFS, skipping resourceExport CR creation for volumes is necessary.
+	// This prevents duplicating restores, as both PVC counts will match after being initiated from driver.StartRestore in Stork context and marking stage as ApplicationRestoreStageApplications
+	// Otherwise, there would be redundant restoration of resources, once from nfs-restore-pvc and another from nfs-restore-resource, potentially marking the restore as partial even in new namespace restores.
+	var onlyPXDdriver bool
+	if len(backupVolumeInfoMappings) == 1 {
+		_, onlyPXDdriver = backupVolumeInfoMappings[volume.PortworxDriverName]
+	}
 	if len(restore.Status.Volumes) != pvcCount {
 		// Here backupVolumeInfoMappings is framed based on driver name mapping, hence startRestore()
 		// gets called once per driver
-		if !nfs {
-			for driverName, vInfos := range backupVolumeInfoMappings {
+		for driverName, vInfos := range backupVolumeInfoMappings {
+			// Portworx driver restore is not supported via job as it can be a secured px volume
+			// to access the token, we need to run the driver.startRestore in the same context as the stork controller
+			// this check is equivalent to (if !nfs || (nfs && driverName == volume.PortworxDriverName))
+			if !nfs || driverName == volume.PortworxDriverName {
```

Comment:
Can you please see if keeping the For loop inside the if !nfs and change the if !nfs {} to if !nfs || hasPxdDriverName {} so that the changes are minimal to handle the case.
You can fetch hasPxdDriverName in the line 618 - 650. 
This wont change the business logic of the code implemented but reduce the amount of wont change that is tested over time.